hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -0,0 +1,273 @@
+    }
+    if (partitionPruningPredicates.nonEmpty) {
+      val predicate = partitionPruningPredicates.reduce(expressions.And)
+
+      val boundPredicate = InterpretedPredicate.create(predicate.transform {
+        case a: AttributeReference =>
+          val index = partitionSchema.indexWhere(a.name == _.name)
+          BoundReference(index, partitionSchema(index).dataType, nullable = true)
+      })
+
+      val partitionPruned = partitionPaths.filter {
+        case PartitionPath(values, _) => boundPredicate.eval(values)
+      }
+      logInfo(s"Total partition size is: ${partitionPaths.size}," +
+        s" after partition prune size is: ${partitionPruned.size}")
+      partitionPruned
+    } else {
+      partitionPaths
+    }
+  }
+
+  /**
+    * Load all partition paths and it's files under the query table path.
+    */
+  private def loadPartitionPathFiles(): Map[PartitionPath, Array[FileStatus]] = {
+    val sparkEngine = new HoodieSparkEngineContext(new JavaSparkContext(spark.sparkContext))
+    val properties = new Properties()
+    properties.putAll(options.asJava)
+    val metadataConfig = HoodieMetadataConfig.newBuilder.fromProperties(properties).build()
+
+    val queryPartitionPath = FSUtils.getRelativePartitionPath(new Path(basePath), queryPath)
+    // Load all the partition path from the basePath, and filter by the query partition path.
+    val partitionPaths = FSUtils.getAllPartitionPaths(sparkEngine, metadataConfig, basePath).asScala
+      .filter(_.startsWith(queryPartitionPath))
+
+    val maxListParallelism = options.get(HoodieWriteConfig.MAX_LISTING_PARALLELISM)
+      .map(_.toInt).getOrElse(HoodieWriteConfig.DEFAULT_MAX_LISTING_PARALLELISM.intValue())
+    val isPartitionedTable = _partitionSchema.fields.isEmpty
+    val serializableConf = new SerializableConfiguration(hadoopConf)
+    val partitionSchema = _partitionSchema
+    val timeZoneId = CaseInsensitiveMap(options)
+      .get(DateTimeUtils.TIMEZONE_OPTION)
+      .getOrElse(SQLConf.get.sessionLocalTimeZone)
+    // List files in all of the partition path.
+    val partition2Files =
+      spark.sparkContext.parallelize(partitionPaths, Math.min(partitionPaths.size, maxListParallelism))
+        .map { partitionPath =>
+          val partitionRow = if (isPartitionedTable) { // This is a none partitioned table
+            InternalRow.fromSeq(Seq(""))
+          } else {
+            val partitionSeqs = partitionPath.split("/")
+            assert(partitionSeqs.length == partitionSchema.size,
+              s"size of partition values[size is: ${partitionSeqs.size}, path " +
+                s"is: '$partitionPath'] is not equal to the size of" +
+                s" partition schema field[size is: ${partitionSchema.size}," +
+                s" fields is: ${partitionSchema.json}].")
+            // Append partition name to the partition value.
+            // e.g. convert "/xx/xx/2021/02" to "/xx/xx/year=2021/month=02"
```

Comment:
We should handle the scenario where `HIVE_STYLE_PARTITIONING_OPT_KEY` was turned on. In this case, the partition path string will already be the required format as I understand.

Reply:
Yes @umehrot2 , If the partition value has already contains the partition name(when  `HIVE_STYLE_PARTITIONING_OPT_KEY`  is enable) I do not append it again. you can see the code bellow:
`val partitionWithName =
            partitionSeqs.zip(partitionSchema).map {
              case (partition, field) =>
                if (partition.indexOf("=") == -1) {
                  s"${field.name}=$partition"
                } else {
                  partition
                }
            }.mkString("/")`